MovieCave - MVVM/Movie-Cave/Movie-Cave/View/MovieScreenVIew/MovieScreenViewController.swift

```diff
@@ -10,44 +10,66 @@ import Combine
 
 class MovieScreenViewController: UIViewController {
 
+    //MARK: - IBOutlets
     @IBOutlet weak var navigationView: NavigationView!
     @IBOutlet weak var movieListView: MovieListView!
     @IBOutlet weak var filterBarView: FilterBarView!
-    private let viewModel = MovieScreenViewModel(movies: [
-        GetMovies.yasuke, GetMovies.dune, GetMovies.rurouniKenshin, GetMovies.guardiansOfGalaxy, GetMovies.oppenheimer, GetMovies.parasite
-    ])
+    
+    //MARK: - Properties
+    private let coreData = CoreDataManager.shared
+    private lazy var viewModel: MovieScreenViewModelProtocol = {
+      return MovieScreenViewModel(coreData: coreData)
+    }()
     private var cancellables: [AnyCancellable] = []
     
+    //MARK: - LifeCycle
     override func viewDidLoad() {
         super.viewDidLoad()
         filterBarView.filterBarDelegate = self
         movieListView.delegate = self
-        
+        navigationView.delegate = self
         viewModel.filteredMovies
             .sink { [weak self] filteredMovies in
             guard let filteredMovies else { return }
             
             self?.movieListView.movies = filteredMovies
         }.store(in: &cancellables)
-        
-        viewModel.searchedMovies
-            .sink { [weak self] filteredMovies in
-            guard let filteredMovies else { return }
+    }
+    
+}
+
+//MARK: - NavigationViewDelegate
+extension MovieScreenViewController: NavigationViewDelegate {
+    
+    func buttonTapped() {
```

Comment:
all the 3 buttons inside this navigation view lead to this view controller but they shoudnt